Common/Base/OpenGL/BaseManager.cpp

```diff
@@ -264,60 +234,56 @@ namespace base
 		}
 	}
 
-	size_t BaseManager::getWindowHandle()
+	void BaseManager::setWindowMaximized(bool _value)
 	{
-		return (size_t)hWnd;
+		if (mWindow != nullptr && _value)
+		{
+			SDL_MaximizeWindow(mWindow);
+		}
 	}
 
-	void BaseManager::setWindowCaption(const std::wstring& _text)
+	bool BaseManager::getWindowMaximized()
 	{
-		SetWindowTextW(hWnd, _text.c_str());
+		Uint32 windowState = SDL_GetWindowFlags(mWindow);
+		return windowState & SDL_WINDOW_MAXIMIZED || windowState & SDL_WINDOW_FULLSCREEN;
 	}
 
-	void BaseManager::prepare()
+	void BaseManager::setWindowCoord(const MyGUI::IntCoord& _value)
 	{
+		if (_value.empty())
+			return;
+
+		MyGUI::IntCoord coord = _value;
+		SDL_Rect screenRect;
+		MYGUI_ASSERT(SDL_GetDisplayBounds(0, &screenRect) != 0, "Failed to get screen size.");
+
+		SDL_SetWindowPosition(mWindow, coord.left, coord.top);
 	}
 
-	void BaseManager::addResourceLocation(const std::string& _name, bool _recursive)
+	MyGUI::IntCoord BaseManager::getWindowCoord()
 	{
-		mPlatform->getDataManagerPtr()->addResourceLocation(_name, _recursive);
+		int left, top, width, height;
+		SDL_GetWindowPosition(mWindow, &left, &top);
+		SDL_GetWindowSize(mWindow, &width, &height);
+		return MyGUI::IntCoord(left, top, width, height);
 	}
 
-	void BaseManager::windowAdjustSettings(HWND hWnd, int width, int height, bool fullScreen)
+	void BaseManager::setWindowCaption(const std::wstring& _text)
 	{
-		// стиль окна
-		HWND hwndAfter = 0;
-		unsigned long style = 0;
-		unsigned long style_ex = 0;
-
-		RECT rc = { 0, 0, width, height };
-
-		if (fullScreen)
-		{
-			style = WS_POPUP | WS_VISIBLE;
-			style_ex = GetWindowLongPtr(hWnd, GWL_EXSTYLE) | (WS_EX_TOPMOST);
-			hwndAfter = HWND_TOPMOST;
+		char mbs[256];				// the title should not be bigger than this array
```

Comment:
We can avoid the fixed-size array by using MyGUI::UString to do the utf8-conversion.